firefox-ios/Client/Frontend/Browser/BackForwardListViewController.swift

```diff
@@ -184,7 +179,8 @@ class BackForwardListViewController: UIViewController,
         with coordinator: UIViewControllerTransitionCoordinator
     ) {
         super.willTransition(to: newCollection, with: coordinator)
-        if shouldShowToolbarForTraitCollection(newCollection) != snappedToBottom, !isBottomSearchBar {
+        let shouldShowToolbar = ToolbarHelper().shouldShowNavigationToolbarForTraitCollection(newCollection)
```

Comment:
could we add this to its own private method that describes why we are doing this logic check? a bit hard for me to understand why we are doing this check without context. my brain can also be mush while reading this PR 🤣  `shouldShowToolbar != snappedToBottom, !isBottomSearchBar`

Reply:
I'll take a look at the code and see how I can make it clearer in my next PR as it will be in the same area anyhow.